src/runtime/hexagon_remote/sim_remote.cpp

```diff
@@ -311,6 +311,18 @@ int initialize_kernels(const unsigned char *code, int codeLen, bool use_dlopenbu
             halide_print(NULL, dlerror());
             return -1;
         }
+
+        if (trace_lib_base) {
+            void *fun_ptr = dlsym(lib, "plt_halide_malloc");
```

Comment:
I'm a bit concerned about the level of hackery that appears to be going on here. What's special about plt_halide_malloc? is it just any old symbol that is found in this library?